### DISCUSSION TITLE
Use poetry for dependencies

### DISCUSSION THREAD

**Discussion by n0emis** [irrdnet/irrd#732] [Ideas]
I have a suggestion to improve dependency handling for this project: Using poetry instead of the manually managed requirements.txt. This would allow to easily lock dependencies but also allow for easier packaging. Also the necessity to specify all dependencies twice (e.g. in the requirements.txt and setup.py) could be removed.

If you a re open this this I could open a PR for this, I just wanted to reach out before...

**Why am I suggesting this?**
I am currently trying to package irrd for nixos, for which either a much more relaxed dependency specification or a complete locking of dependencies would be required.

**Comment by mxsasha** [comment]
I like this idea. Thanks for suggesting! It's been a minor annoyance for a while, especially having to keep dependencies up to date in two places, but it hasn't annoyed me enough to fix it. Poetry seems the standard now.

For version pinning, I strongly prefer locking exact versions, for reproducibility. This is a big enough change that I'd want to hold it back for 4.4, i.e. not include it in the upcoming 4.3 release.

**Comment by mxsasha** [comment]
It's done, #760. Installation dependencies are still not very relaxed though, because poetry doesn't seem to have a way to create the package from the lock file, at least not yet, which would be more suitable for IRRD as it's mostly an application.
